core/src/main/scripts/importer/validateData.py

```diff
@@ -3229,87 +3263,228 @@ def parse_q_value(self, value, column_number):
         else:
-            # Check if this is the second gene set data file
-            if prior_validated_geneset_ids is not None:
-                # Check if gene set is in the first gene set data file
-                if geneset_id not in prior_validated_geneset_ids:
-                    self.logger.error('Gene sets in cannot be found in other gene set file',
+            # Check if this is the second treatment data file
+            if self._get_prior_validated_feature_ids() is not None:
+                # Check if treatment is in the first treatment data file
+                if feature_id not in self._get_prior_validated_feature_ids():
+                    self.logger.error('Feature id cannot be found in other data file',
                                       extra={'line_number': self.line_number,
-                                             'cause': geneset_id})
-            # Add gene set to list of gene sets of current gene set data file
-            self.geneset_ids.append(geneset_id)
-        return geneset_id
+                                             'cause': feature_id})
+            # Add treatment to list of treatments of current treatment data file
+            self.feature_ids.append(feature_id)
+        return feature_id
 
     def onComplete(self):
 
-        def checkConsistencyScoresPvalue(self):
-            """This function validates whether the gene sets in the scores and p-value file are the same"""
-
-            global prior_validated_geneset_ids
+        def checkConsistencyFeatures(self):
+            """This function validates whether the treatments in the treatment response files (IC50, EC50, 
+            GI50, AUC, ...) are the same"""
 
-            # If the prior_validated_geneset_ids is not filled yet, fill it with the first file.
-            if prior_validated_geneset_ids is None:
-                prior_validated_geneset_ids = self.geneset_ids
+            # If the prior_validated_features_ids is not filled yet, fill it with the first file.
+            if self._get_prior_validated_feature_ids() is None:
+                self._set_prior_validated_feature_ids(self.feature_ids)
             else:
-                # Check if gene set ids are the same
-                if not prior_validated_geneset_ids == self.geneset_ids:
-                    self.logger.error(
-                        'Gene sets column in score and p-value file are not equal')
+                # Check if feature ids are the same
+                if not self._get_prior_validated_feature_ids() == self.feature_ids:
+                    self.logger.error( self._get_message_features_do_not_match() )
 
-        checkConsistencyScoresPvalue(self)
+        checkConsistencyFeatures(self)
 
-        super(GsvaWiseFileValidator, self).onComplete()
+        super(MultipleDataFileValidator, self).onComplete()
+
+# TODO: move duplicate code in GsvaWiseFileValidator and TreatmentWiseFileValidator classes to meta class
```

Comment:
Do you mean moving them to the superclass? I see no reason for metaprogramming there.

Reply:
Comment was no longer needed. Deleted.

Reply:
Now that I think about it again, this may actually have been a sensible idea 😳  But if you don't feel like it, I'd say it's fine not to do it right away.

Reply:
Sorry, I must have been really out of practice here. Trying to solve this with metaclasses would be a terrible idea—those are about inheritance. If anything, the feature that could be used for that cleanup would be `@`decorators.